File: tls/tls-java/app/src/main/java/com/gmail/volodymyrdotsenko/tls/RequestHandler.java
package com.gmail.volodymyrdotsenko.tls;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

public final class RequestHandler implements Runnable {
    private static final Logger LOG = LoggerFactory.getLogger(RequestHandler.class);

    private final Service service;
    private final List<Request> requests;

    public RequestHandler(Service service, List<Request> requests) {
        this.service = service;
        this.requests = List.copyOf(requests);
    }

    @Override
    public void run() {
        LOG.info("Starting request handler with {} requests", requests.size());
        try {
            for (Request request : requests) {
                LOG.info("Sending request: {}", request);
                var response = service.get(request);
                LOG.info("Got response: {}", response);
            }
        } finally {
            service.clearContext();
        }
    }
}

File: tls/tls-rust/src/request_handler.rs
use crate::request::Request;
use crate::response::ResponseStatus;
use crate::service_v2::Service;
use tracing::{Level, event};

pub struct RequestHandler {
    service: Service,
    requests: Vec<Request>,
}

impl RequestHandler {
    pub fn new(service: Service, requests: Vec<Request>) -> Self {
        Self { service, requests }
    }

    pub fn run(&self) {
        event!(
            Level::INFO,
            "Starting request handler with {} requests",
            self.requests.len()
        );
        for request in &self.requests {
            event!(Level::INFO, "Sending request: {}", request);
            let response = self.service.get(request);
            match response.status {
                ResponseStatus::Success => event!(Level::INFO, "Got response: Success"),
                ResponseStatus::SuccessAlreadyLoggedIn => {
                    event!(Level::INFO, "Got response: SuccessAlreadyLoggedIn")
                }
                ResponseStatus::AuthError => println!("Got response: AuthError"),
            }
        }
    }
}
